### ISSUE TITLE
Some doxygen links to cantera.org are broken

### ISSUE THREAD

**Issue by ischoegl** [Cantera/cantera#1592]
**Problem description**

The top menu "cantera.org" item on the updated doxygen website links back to the main documentation. From many pages things work as expected (e.g. from the [main index page](https://cantera.org/documentation/dev/doxygen/html/index.html)), with the intended redirect

```
https://cantera.org/
```

On other pages, the html code gets scrambled (e.g. [deprecated list](https://cantera.org/documentation/dev/doxygen/html/da/d58/deprecated.html)), i.e

```html
https://cantera.org/documentation/dev/doxygen/html/https://cantera.org
```

**Behavior**

**System information**

- Cantera version: `main`
- OS: N/A

**Additional context**

The XML entry generating the link is https://github.com/Cantera/cantera/blob/cde1e79567519646d1b6b9520c4a24d4f22f0f6c/doc/doxygen/DoxygenLayout.xml#L37

This follows the [doxygen documentation](https://www.doxygen.nl/manual/customize.html#layout)
```
  <navindex>
    ...
    <tab type="user" url="http://www.google.com" title="Google"/>
    ...
  </navindex>
```

Other entries here have an additional ` intro=""`, which may or may not make a difference ... at the same time, this may be an upstream issue.

**Comment by speth** [comment]
Looks like an upstream issue (https://github.com/doxygen/doxygen/issues/9074) that was already fixed in Doxygen 1.9.4.

**Comment by ischoegl** [comment]
Thanks for tracking this down! I believe the runner builds with 1.9.1 - I guess `ubuntu-latest` (Ubuntu 22.04) uses an earlier version (1.9.1 was released Dec 24, 2022) ... I guess we can either use a newer pip version or wait until Ubuntu 24.04 makes it to GitHub? It's definitely a paper cut ...
